hypershift-operator/controllers/hostedcluster/internal/platform/aws/aws.go

```diff
@@ -43,7 +43,12 @@ func (p AWS) ReconcileCAPIInfraCR(hcluster *hyperv1.HostedCluster, controlPlaneN
 	if err != nil {
 		return nil, err
 	}
-
+	// reconciliation strips TypeMeta. We repopulate the static values since they are necessary for
+	// downstream reconciliation of the CAPI Cluster resource.
+	awsCluster.TypeMeta = metav1.TypeMeta{
+		Kind:       "AWSCluster",
+		APIVersion: capiawsv1.GroupVersion.String(),
```

Comment:
It actually looks fine in my current cluster
```
  infrastructureRef:
    apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
    kind: AWSCluster
    name: agl
    namespace: clusters-agl
```

Reply:
We use a gvk find previously

Reply:
it always matches v1beta1 if you use gvk find and doesn't properly support multiple versions